java-client/src/main/java/org/opensearch/client/transport/aws/AwsSdk2Transport.java

```diff
@@ -615,4 +622,86 @@ private static <T> Optional<T> or(Optional<T> opt, Supplier<? extends Optional<?
             return Objects.requireNonNull(r);
         }
     }
+
+    /**
+     * Wrap the exception so the caller's signature shows up in the stack trace, taking care to copy the original type and message
+     * where possible so async and sync code don't have to check different exceptions.
+     */
+    private static Exception extractAndWrapCause(Exception exception) {
+        if (exception instanceof InterruptedException) {
+            throw new RuntimeException("thread waiting for the response was interrupted", exception);
+        }
+        if (exception instanceof ExecutionException) {
+            ExecutionException executionException = (ExecutionException) exception;
+            Throwable t = executionException.getCause() == null ? executionException : executionException.getCause();
+            if (t instanceof Error) {
+                throw (Error) t;
+            }
+            exception = (Exception) t;
+        }
+        if (exception instanceof ConnectTimeoutException) {
```

Comment:
AFAIK AWS SDK does not support ACH 5 (only ACH 4 as per https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/http-configuration.html), how come we're seeing this exceptions?

Reply:
Thank you